core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -374,6 +374,10 @@ class KafkaApis(val requestChannel: RequestChannel,
    */
   def handleProduceRequest(request: RequestChannel.Request, requestLocal: RequestLocal): Unit = {
     val produceRequest = request.body[ProduceRequest]
+    if (produceRequest.version <= 2) {
```

Comment:
should there be a test/documentation explaining why this is done? otherwise could easily get swept away in a future change

Reply:
As per the PR description, this is WIP - the code will change before review time. :)

Reply:
I have updated the PR.